src/parsers/power_models_data.jl

```diff
@@ -427,33 +427,41 @@ function make_thermal_gen(gen_name::AbstractString, d::Dict, bus::ACBus, sys_mba
         # Input data layout: table B-4 of https://matpower.org/docs/MATPOWER-manual.pdf
         if model == GeneratorCostModels.PIECEWISE_LINEAR
             # For now, we make the fixed cost the y-intercept of the first segment of the
-            # piecewise curve and the variable cost a PiecewiseLinearPointData representing
+            # piecewise curve and the variable cost a PiecewiseLinearData representing
             # the data minus this fixed cost; in a future update, there will be no
-            # separation between the PiecewiseLinearPointData and the fixed cost.
+            # separation between the PiecewiseLinearData and the fixed cost.
             cost_component = d["cost"]
             power_p = [i for (ix, i) in enumerate(cost_component) if isodd(ix)]
             cost_p = [i for (ix, i) in enumerate(cost_component) if iseven(ix)]
             points = collect(zip(power_p, cost_p))
             (first_x, first_y) = first(points)
             fixed = max(0.0,
-                first_y - first(get_slopes(PiecewiseLinearPointData(points))) * first_x,
+                first_y - first(get_slopes(PiecewiseLinearData(points))) * first_x,
             )
-            cost = PiecewiseLinearPointData([(x, y - fixed) for (x, y) in points])
+            cost = PiecewiseLinearData([(x, y - fixed) for (x, y) in points])
         elseif model == GeneratorCostModels.POLYNOMIAL
-            # For now, we make the variable cost a PolynomialFunctionData with all but the
+            # For now, we make the variable cost a QuadraticFunctionData with all but the
             # constant term and make the fixed cost the constant term; in a future update,
-            # there will be no separation between the PolynomialFunctionData and the fixed
+            # there will be no separation between the QuadraticFunctionData and the fixed
             # cost.
             # This transforms [3.0, 1.0, 4.0, 2.0] into [(1, 4.0), (2, 1.0), (3, 3.0)]
             coeffs = enumerate(reverse(d["cost"][1:(end - 1)]))
-            cost = PolynomialFunctionData(Dict((i, c / sys_mbase^i) for (i, c) in coeffs))
+            coeffs = Dict((i, c / sys_mbase^i) for (i, c) in coeffs)
+            quadratic_degrees = [2, 1, 0]
+            (keys(coeffs) <= Set(quadratic_degrees)) || throw(
```

Comment:
Is this idiomatic Julia? It also seems more natural to me to say `if error-condition throw` or `error-condition && throw`. Asserts use this form.